package pb;

import "resource.proto";

message Server {
    extend Resource {
        required Server resource = 101;
    }

    message Storage {
        optional int32        id           = 1;
        required string       name         = 2;
        required string       url          = 3;
        required int64        spaceLimit   = 4;
    }

    required string       apiUrl       = 1;
    optional string       netAddrList  = 2;
    optional bool         reserve      = 3;
    repeated Storage      storage      = 4;
}
